From 6545e6a55d4aaad0f48f2bb5c865935692ce32ea Mon Sep 17 00:00:00 2001
From: Damien Miller <djm@mindrot.org>
Date: Fri, 16 Oct 2026 17:13:41 +1100
Subject: [PATCH] another readpassphrase() fix from OpenBSD libc

revision 1.32
date: 2026/10/15 16:40:50;  author: millert;  state: Exp;  lines: +2 -2;  commitid: PeYLMhPn3x9iWE37;
Don't print newline after reading password if interrupted by signal.
Avoids an extra newline when the user presses ^C or ^\ at the
password prompt.  OK deraadt@

(cherry picked from commit 6545e6a55d4aaad0f48f2bb5c865935692ce32ea)
---
 openbsd-compat/readpassphrase.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/openbsd-compat/readpassphrase.c b/openbsd-compat/readpassphrase.c
index 4b710b4..d5fff87 100644
--- a/openbsd-compat/readpassphrase.c
+++ b/openbsd-compat/readpassphrase.c
@@ -1,4 +1,4 @@
-/*	$OpenBSD: readpassphrase.c,v 1.31 2026/10/15 05:55:06 djm Exp $	*/
+/*	$OpenBSD: readpassphrase.c,v 1.32 2026/10/15 16:40:50 millert Exp $	*/
 
 /*
  * Copyright (c) 2000-2002, 2007, 2010
@@ -162,7 +162,7 @@ restart:
 	}
 	*p = '\0';
 	save_errno = errno;
-	if (!(term.c_lflag & ECHO))
+	if (!(term.c_lflag & ECHO) && !(nr == -1 && errno == EINTR))
 		(void)write(output, "\n", 1);
 
 	/* Restore old terminal settings and signals. */
-- 
2.53.0

